Kitodo/src/main/java/de/sub/goobi/forms/BenutzergruppenForm.java

```diff
@@ -136,6 +139,41 @@ public int getUserGroupId() {
         return this.userGroupId;
     }
 
+    /**
+     * Return the list of available authorization levels and the list of
+     * authorization levels currently assigned to 'myBenutzergruppe' as a combined
+     * 'DualListModel' that is used by the frontend for authorization management of
+     * user groups utilizing a PrimeFaces PickList object.
+     *
+     * @return DualListModel of available and assigned authorization levels
+     */
+    public DualListModel<Authorization> getAuthorizations() {
+        List<Authorization> assignedAuthorizations = this.myBenutzergruppe.getAuthorizations();
+        List<Authorization> availableAuthorizations = serviceManager.getAuthorizationService().getAll();
+        availableAuthorizations.removeAll(assignedAuthorizations);
+        return new DualListModel<>(availableAuthorizations, assignedAuthorizations);
+    }
+
+    /**
+     * Assign the target property of given DualListModel of authorizations to
+     * 'myBenutzergruppe' using a PrimeFaces PickList object.
+     *
+     * @param authorizations
+     *            list of authorizations assigned to 'myBenutzergruppe'
+     */
+    public void setAuthorizations(DualListModel<Authorization> authorizations) {
+        try {
+            List<Authorization> assignedAuthorizations = new ArrayList<>();
+            for (Object object : authorizations.getTarget()) {
+                assignedAuthorizations
+                        .add(serviceManager.getAuthorizationService().getById(Integer.parseInt((String) object)));
```

Comment:
This looks ugly to me as you converting a list entry from `Object` to `String` and then you hope that this could be used as an `Integer` value. Is there no other way for converting and retrieving the necessary data? As there are no test which are proving your assumption made me the decision not easy.